doc/vxlan/EVPN/EVPN_VXLAN_HLD.md

```diff
@@ -0,0 +1,1023 @@
+
+### 4.1.3 SAI Overview
+
+To support this feature, SAI will be extended as below:
+
+- **sai_fdb_entry_type_t** - will be enhanced to have a new enum. 
+  - SAI_FDB_ENTRY_TYPE_STATIC_MACMOVE - This will be used when a MAC should not be aged out but a MAC move should be allowed. An example is remote MAC learnt by EVPN procedures which can be subjected to a MAC move from remote to local.
+- **sai_tunnel_attr_t** - It will be enhanced to have one new attribute
+  - SAI_TUNNEL_ATTR_ENCAP_DEST_IP - New tunnel attribute will be added to specify the tunnel destination IP address. This will be used to support warm reboot. Currently only the SIP is part of the tunnel attributes which does not allow for unique set of attributes when there are multiple tunnels with same SIP and different DIPs. The unique set of attributes is required for reconcilation during a warm reboot.  bridgeport associated with a tunnel also requires the tunnel id to be reconcilable post warm restart. Hence, it is required as part of the sai_tunnel_attr_t list.
+- **sai_vlan_attr_t** - It will be enhanced to have one new attribute
+  - SAI_VLAN_ATTR_NEIGH_SUPP_MODE - By default this value will be FALSE. Reason for this new attribute is due to recent changes in ARP behaviour. The default arp behavior in copp file is changed from trap to copy(as part of commit -5e4b71d4). But for ARP Suppression, we need Pkt to get trapped instead of copy to avoid flooding of packets over vxlan tunnel.So a differentiated tcam rule for these arp suppressed vlans will redirect the ARP pkts to cpu instead of copy.
+
+The changes are subject to approval by the SAI community.
+
+## 4.2 DB Changes
+
+At a high level below are some of the interactions between relevant components and the DB involved for EVPN L2 VXLAN support in SONiC architecture.
+
+![EVPN L2 VXLAN Arch](images/arch.PNG "Figure : Arch")
+__Figure 4: EVPN L2 VXLAN DBs__
+
+### 4.2.1 CONFIG_DB changes
+
+**EVPN_NVO_TABLE**
+
+Producer:  config manager 
+
+Consumer: VxlanMgr
+
+Description: This is a singleton configuration which holds the VTEP source IP used for BGP-EVPN discovered tunnels. 
+
+Schema:
+
+```
+;New table
+;holds the VTEP source IP used for BGP-EVPN discovered tunnels
+
+key = EVPN_NVO_TABLE:nvo_name
+                          ; nvo or VTEP name as a string
+; field = value
+source_vtep = tunnel_name ; refers to the tunnel object created with SIP only.
+```
+
+**NEIGH_SUPPRESS_CFG_VLAN_TABLE**
+
+Producer:  config manager
+
+Consumer: VlanMgr
+
+Description: New table that stores neighbor suppression per VLAN configuration.
+
+Schema:
+
+```
+;New table
+;Stores Neigh Suppress configuration per VLAN
+
+key             = NEIGH_SUPPRESS_VLAN:"Vlan"vlanid       ; VLAN with prefix "NEIGH_SUPPRESS"
+suppress_neigh  = "ON" / "OFF" ; Default "OFF"
+
+```
+
+**VRF_TABLE**
```

Comment:
There is already a L3 VxLAN implementation already in SONiC called VNET. Why not integrate with it?

Reply:
EVPN L3VNI design here is already integrated with VXLAN design/implementation done as part of VNET. Most of the work is being leveraged. However, since VNET is not VRF, and VRF support was added later into SONiC, it makes more sense to provide L3VNI solution with VRF.
VNET will continue to co-exist with VRF-L3VNI.

Reply:
Can both VNET based tunnel and VRF based tunnel be created simultaneously on a switch?  If not, any warning message when a user does so (mostly by mistake, say when migrating from VNET to VRF)?

Reply:
VNET co-exist with VRF-L3VNI. Currently no such checks.